js2-mode.el

```diff
@@ -10365,11 +10392,14 @@ Last token parsed must be `js2-RB'."
   (when (eq (js2-token-type token) js2-NAME)
     (js2-record-face 'js2-function-call token)))
 
-(defun js2-parse-function-call (pn)
+(defun js2-parse-function-call (pn &optional use-optional-chaining-p)
   (js2-highlight-function-call (js2-current-token))
   (js2-get-token)
   (let (args
         (pos (js2-node-pos pn)))
+    (when use-optional-chaining-p
```

Comment:
How come this is needed for optional chaining, but not for normal chaining?

Reply:
See https://babeljs.io/docs/en/babel-plugin-proposal-optional-chaining

One example is `Test?.();` instead of `Test()`

So `Test` is still a function call token but there might exist a optional chaining token between `Test` and `(params)`.

AST is like 

```
                                 function call token
                             /                |                               \
function name token   optional chaining token  paramters token
```